devito/passes/iet/definitions.py

```diff
@@ -276,7 +276,7 @@ def _alloc_pointed_array_on_high_bw_mem(self, site, obj, storage):
 
         memptr = VOID(Byref(obj._C_symbol), '**')
         alignment = obj._data_alignment
-        nbytes = SizeOf(Keyword('%s*' % obj._C_typedata))*obj.dim.symbolic_size
+        nbytes = SizeOf(obj._C_typedata, stars='*')*obj.dim.symbolic_size
```

Comment:
oh I see why you have a special handler for SizeOf ... you've added a new attribute `stars`. I think, in the future, the bettwe way to go could be that of adding an object such as `Star(...)` or `Dereference(...)` or (maybe definitive) `Type(..., stars=...)`

Reply:
It's not just adding the stars, it does not turn the type directly to a string because it needs to generate the right string depending on the printer/languagage